### DISCUSSION TITLE
OAuth: Add auto user registration on first login

### DISCUSSION THREAD

**Discussion by saph1s** [pelican-dev/panel#758] [Ideas]
At this moment (v1.0.0-beta14) there is no solution for auto user registration on login via OAuth (Authentik as example).

It would be nice to have this feature!

**Comment by Boy132** [comment]
Very unlikely.
Admins still need to assign servers to the user so they can actually do something.

**Reply by rmartinoscar** [reply]
Could be cool for subusers but its not a must-have.
As for user registration it should be optional (and disabled by default)

**Reply by saph1s** [reply]
> Very unlikely. Admins still need to assign servers to the user so they can actually do something.

Yeah, I understand that. But you don't need to create user by yourself. 
Also, auto role assignment by claims can simplify the process significantly. With auto role assignment based on claims, users can be granted appropriate permissions automatically during the login process, reducing the workload on admins. This approach streamlines user onboarding while still allowing admins to manage server assignments as a separate step when needed.

**Comment by Boy132** [comment]
Will be included in the next beta, see #1573!
